src/main/java/Group15/View/MuscleSelectionView.java

```diff
@@ -232,13 +289,47 @@ private static void updateBodyPartLists(BodyPart bodyPart, BodyPartButtonStates
     }
 
     private static void createSubmitButtonFunctionality(TextField minutesInputField) {
+        MuscleSelectionView.errorList.clear();
+
         if (minutesInputField.getText() == null || minutesInputField.getText().isEmpty() || !minutesInputField.getText().matches("\\d+")) {
-            System.out.println("Invalid input");
+            MuscleSelectionView.errorList.add("Invalid Input for Time!");
+            updateErrorMessageTextArea();
+        }
+        if(selectedBodyParts.isEmpty()) {
+            MuscleSelectionView.errorList.add("No Bodyparts Selected!");
+            updateErrorMessageTextArea();
+        }
+        if(selectedEquipment.isEmpty()) {
+            MuscleSelectionView.errorList.add("No Equipment Selected");
+            updateErrorMessageTextArea();
+        }
+        if(!errorList.isEmpty()) {
+            updateErrorMessageTextArea();
             return;
         }
 
         int timeInMinutes = Integer.parseInt(minutesInputField.getText());
 
+        Dialog<String> workoutNameDialog = createWorkoutNameDialog();
+
+        workoutNameDialog.showAndWait().ifPresentOrElse(
+                workoutName -> {
+                    String finalWorkoutName = workoutName.trim().isEmpty() ? "Workout" : workoutName.trim();
+                    MuscleSelectionView.setWorkoutName(finalWorkoutName);
+                    ViewController.setScene(WorkoutView.createScene(
+                            WorkoutAlgorithm.createWorkoutFromExercises(selectedBodyParts, dislikedBodyParts, selectedEquipment, timeInMinutes)
+                    ));
+                },
+                () -> {
+                    MuscleSelectionView.setWorkoutName("Workout");
+                    ViewController.setScene(WorkoutView.createScene(
```

Comment:
why the second one? if the workout name is empty the first one looks like it sets the name to "Workout". So it looks like you are creating two workouts